clang/tools/3c/3CStandalone.cpp

```diff
@@ -250,6 +250,20 @@ static cl::opt<bool> OptForceItypes(
     cl::init(false), cl::cat(_3CCategory));
 #endif
 
+std::string getFinalStatsPath(const std::string &J) {
+  // Get final path for a stats file.
+  SmallString<255> AbsPath;
+  std::error_code EC = llvm::sys::fs::real_path(J, AbsPath);
+  // If the path is not absolute path, then append output directory.
+  if (!EC && J != AbsPath.str()) {
```

Comment:
In the corner case that `J` is something like `/home/mwh/util/../output` then even though it's an output path, it will not be the case that `J == AbsPath.str()`. Use [llvm::sys::path::is_relative()](https://llvm.org/doxygen/namespacellvm_1_1sys_1_1path.html#a59d172f36ecf079548e9c539ae54e5a4) instead? Likewise, for what's below, use `llvm::sys::path::` operations to construct a path directly, and then get a string from it, rather than constructing the string directly. I'm a little worried about Windows.